### DISCUSSION TITLE
Limit on number of preloaded images

### DISCUSSION THREAD

**Discussion by SameerSabharwalSiddiqi** [jspsych/jsPsych#1483] [Q&A]
Dear JS"psychers",

I'm programming an associative memory task that requires quite a few images to be uploaded onto cognition.run (n=~700). 

As would perhaps be expected, I'm running into RAM issues. The program doesn't crash, but since images are programmed to be rapidly presented during a masking phase, some images simply won't load (so I see a white screen in lieu of a masking presentation). Interestingly, this problem persists in Chrome but not in Firefox - and I'm using windows.

So while this problem is perhaps more appropriate for a general programming/JS forum, I am wondering if anyone else has tried programming experiments with > 500 images (~30kb size of each image). Were you able to overcome these sorts of memory issues? Is there on the other hand a hard-limit to the amount of images that one is able to load onto cognition.run? 

Thank you all ! 
Sameer

**Comment by jodeleeuw** [comment]
Hi Sameer,

I think you are running into just a general limitation on browser caching that jsPsych has limited control over. Or at least I haven't been able to figure out ways of creating more robust preloading methods yet!

We are making some changes to how we implement preloading in version 6.3, which will be out within the next week or so. We're introducing a plugin to do preloading, so that you can preload portions of your experiment at a time. That might help quite a bit in this situation. You could preload the images you need for a block of trials, and then preload the next batch for the next block, etc.

Cheers,
Josh

**Reply by SameerSabharwalSiddiqi** [reply]
Josh, 

The new plug-in sounds very promising. Yes, I do believe that this is a general browser-caching problem (as evidenced by it working on some browsers, firefox, but not others, chrome). However, being able to preload media in portions would certainly take a load of the memory and allow for smoother operations on all devices. So sincere thanks for the heads-up, the update, and look forward to trying it out in the near future.

:), Sameer
